compiler/src/dotty/tools/repl/ReplCompiler.scala

```diff
@@ -88,6 +88,14 @@ class ReplCompiler extends Compiler {
     var valIdx = state.valIndex
     val defs = new mutable.ListBuffer[Tree]
 
+    def resNum(name: Name) = name.show.stripPrefix(str.REPL_RES_PREFIX).toIntOption
+
+    def maybeBumpValIdx(tree: Tree): Unit = tree match
```

Comment:
Can you document why this is needed and point the reader to a test case since it's non-obvious?